src/lib/pgp-key.cpp

```diff
@@ -1596,7 +1590,9 @@ pgp_key_t::lock()
         return true;
     }
 
-    material().clear_secret();
+    if (material()) {
```

Comment:
maybe it will be good to add assert not null here

Reply:
Thanks, addressed this and the next comments.